src/fragments/lib-v1/storage/js/upload.mdx

```diff
@@ -1,173 +1,189 @@
 
-When a page refresh occurs during the upload, re-initializing the upload with the same file will continue from previous progress point.
-
-```javascript
-const upload = Storage.put(file.name, file, {
-  resumable: true,
-});
+<!-- TODO Rewrite why do we not have multi-part upload documented for JS? -->
 
+When a page refresh occurs during the upload, re-initializing the upload with the same file will continue from previous progress point.
 
-// This duplicate uploads will resume the original upload.
-const duplicateUpload = Storage.put(file.name, file, {
-  resumable: true,
-});
-```
 <Callout>
 
 Uploads that were initiated over one hour ago will be cancelled automatically. There are instances (e.g device went offline, user logs out) where the incomplete file remains in your S3 account. It is recommended to [setup a s3 lifecycle rule](https://aws.amazon.com/blogs/aws-cloud-financial-management/discovering-and-deleting-incomplete-multipart-uploads-to-lower-amazon-s3-costs/) to automatically cleanup incomplete upload requests.
 
 </Callout>
 
-### Event handlers
-With the `resumable: true` flag, there are 3 callback functions available: `completeCallback`, `progressCallback` and `errorCallback`.
-```javascript
-const upload = Storage.put(file.name, file, {
-		resumable: true,
-		completeCallback: (event) => {
-			console.log(`Successfully uploaded ${event.key}`);
-		},
-		progressCallback: (progress) => {
-			console.log(`Uploaded: ${progress.loaded}/${progress.total}`);
-		},
-		errorCallback: (err) => {
-			console.error('Unexpected error while uploading', err);
-		}
-})
-```
-
-## Browser uploads
-
-Upload an image in the browser:
-
-```javascript
-async function onChange(e) {
-  const file = e.target.files[0];
-  try {
-    await Storage.put(file.name, file, {
-      contentType: "image/png", // contentType is optional
-    });
-  } catch (error) {
-    console.log("Error uploading file: ", error);
-  }
-}
-
-<input type="file" onChange={onChange} />;
-```
-
 <Callout>
 
-Note: 'contentType' is metadata (saved under the key 'Content-Type') for the S3 object and does not determine the 'Type' in the AWS S3 Console. If a file extension is not provided in the key of the uploaded object, the S3 console's 'Type' field will be omitted. Otherwise, the 'Type' will be populated to match the given extension of the key. The behavior of how the S3 object is treated will be based on 'contentType' in the metadata and not the 'Type'. 
-
-For example: uploading a file with the key "example.jpg" will result in the 'Type' being set as "jpg", but the 'contentType' in metadata will determine it's behavior so setting it as "text/html" will result in the file being treated as an HTML file regardless of displayed 'Type' in the S3 console.
+Currently, the `onProgress` event is dependent and based on the number of parts that are sent to S3. Currently, the library uploads at about 5 MB per part. If you upload a file smaller than this, the event will not be fired.
```

Comment:
Is this true? the progress event should be fired regardless of the 5 MB limit. The progress is dirrectly emitted from the underlying HTTP payload.